tide_site_restriction.module

```diff
@@ -333,3 +334,36 @@ function tide_site_restriction_form_user_pass_reset_alter(&$form, FormStateInter
 function _user_login_page($form, $form_state) {
   $form_state->setRedirect('<front>');
 }
+
+/**
+ * Implements hook_entity_operation_alter().
+ */
+function tide_site_restriction_entity_operation_alter(array &$operations, EntityInterface $entity) {
+  if (isset($operations['quick_clone'])) {
+    /** @var \Drupal\tide_site_restriction\Helper $site_restriction_helper */
+    $site_restriction_helper = \Drupal::service('tide_site_restriction.helper');
+    $user = User::load(\Drupal::currentUser()->id());
+    $user_sites = $site_restriction_helper->getUserSites($user);
+    $user_can_bypass_restriction = $site_restriction_helper->canBypassRestriction(\Drupal::currentUser());
+    if (!$user_can_bypass_restriction) {
```

Comment:
@vincent-gao just for curiosity does it still needs to check the following condition if a user doesn't have the permission to bypass the restriction? My thought is if it doesn't have the access to bypass, it shouldn't have the permission to clone.

Reply:
@MdNadimHossain it based on what sites the user has, so it needs following conditions.